### ISSUE TITLE
FileNotFoundException with spaces in path to user profile

### ISSUE THREAD

**Issue by Fleex255** [robolectric/robolectric#4589]
### Description
When the path to the current Windows user's profile folder contains a space (e.g. `C:\Users\John Smith`), trying to run Robolectric tests will fail:

    java.lang.RuntimeException: java.io.FileNotFoundException: C:\Users\John%20Smith\.m2\repository\org\robolectric\android-all\9-robolectric-4913185-2\android-all-9-robolectric-4913185-2.jar (The system cannot find the path specified)

It appears that the path is being inappropriately URL-encoded, replacing spaces with `%20`. We have worked around the problem by creating a symbolic link from e.g. `John%20Smith` to `John Smith`, but this is obviously not ideal.

Possibly related to https://github.com/robolectric/robolectric/issues/2171.

### Steps to Reproduce
1. Log into Windows as a user whose profile folder name contains a space
2. Run any Robolectric test suite in Android Studio

### Robolectric & Android Version
Robolectric 4.1, Android API level 28, JUnit 4.12, Windows 10

### Link to a public git repo demonstrating the problem:
The problem does not appear to be related to any specific Robolectric-based test suite.

**Comment by xian** [comment]
@Fleex255 — can you provide a full stacktrace?

**Comment by Fleex255** [comment]
Here it is:

```
java.lang.RuntimeException: java.io.FileNotFoundException: C:\Users\Space%20Test\.m2\repository\org\robolectric\android-all\9-robolectric-4913185-2\android-all-9-robolectric-4913185-2.jar (The system cannot find the path specified)

	at org.robolectric.res.Fs$JarFs.<init>(Fs.java:101)
	at org.robolectric.res.Fs.fromJar(Fs.java:27)
	at org.robolectric.internal.SdkEnvironment.createRuntimeSdkResourcePath(SdkEnvironment.java:43)
	at org.robolectric.internal.SdkEnvironment.getSystemResourceTable(SdkEnvironment.java:34)
	at org.robolectric.ApkLoader.getSystemResourceTable(ApkLoader.java:32)
	at org.robolectric.android.internal.ParallelUniverse.injectResourceStuffForLegacy(ParallelUniverse.java:252)
	at org.robolectric.android.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:126)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:379)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:252)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:130)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:84)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.io.FileNotFoundException: C:\Users\Space%20Test\.m2\repository\org\robolectric\android-all\9-robolectric-4913185-2\android-all-9-robolectric-4913185-2.jar (The system cannot find the path specified)
	at java.util.zip.ZipFile.open(Native Method)
	at java.util.zip.ZipFile.<init>(ZipFile.java:219)
	at java.util.zip.ZipFile.<init>(ZipFile.java:149)
	at java.util.jar.JarFile.<init>(JarFile.java:166)
	at java.util.jar.JarFile.<init>(JarFile.java:130)
	at org.robolectric.res.Fs$JarFs.<init>(Fs.java:99)
	... 22 more
```

**Comment by xian** [comment]
Confirmed fixed in 4.2.1 (release forthcoming).

**Comment by xian** [comment]
4.2.1 is released: https://github.com/robolectric/robolectric/releases/tag/robolectric-4.2.1
